samples/siwx91x_ota/src/main.c

```diff
@@ -0,0 +1,838 @@
+				   struct net_if *iface)
+{
+	char addr[INET_ADDRSTRLEN];
+	struct app_ctx *ctx = CONTAINER_OF(cb, struct app_ctx, ipv4_mgmt_cb);
+
+	net_addr_ntop(AF_INET, cb->info, addr, sizeof(addr));
+	if (mgmt_event == NET_EVENT_IPV4_ADDR_ADD) {
+		printf("Added IPv4 Address: %s\n", addr);
+		ctx->ipv4_addr_config_done = true;
+		k_event_post(&ctx->events, NET_IPV4_EVENT_READY);
+	} else {
+		ctx->ipv4_addr_config_done = false;
+		printf("Removed IPv4 Address: %s\n", addr);
+		ctx->state = OTA_STATE_SCAN;
+		ota_cleanup_resources(ctx);
+	}
+}
+
+/*
+ * @brief IPv6 event handler for address add/remove events
+ *
+ * @param cb Pointer to the event callback structure
+ * @param event The network event to handle
+ * @param iface Network interface that generated the event
+ */
+static void ota_ipv6_event_handler(struct net_mgmt_event_callback *cb, uint64_t mgmt_event,
+				   struct net_if *iface)
+{
+	char addr[INET6_ADDRSTRLEN];
+	struct app_ctx *ctx = CONTAINER_OF(cb, struct app_ctx, ipv6_mgmt_cb);
+
+	net_addr_ntop(AF_INET6, cb->info, addr, sizeof(addr));
+	if (mgmt_event == NET_EVENT_IPV6_ADDR_ADD) {
+		printf("Added IPv6 Address: %s\n", addr);
+		k_event_post(&ctx->events, NET_IPV6_EVENT_READY);
+	} else {
+		printf("Removed IPv6 Address: %s\n", addr);
+		ctx->state = OTA_STATE_SCAN;
+		ota_cleanup_resources(ctx);
+	}
+}
+
+static void ota_dns_event_handler(struct net_mgmt_event_callback *cb, uint64_t mgmt_event,
+				  struct net_if *iface)
+{
+	const struct sockaddr *entry = cb->info;
+	char addr[INET6_ADDRSTRLEN];
+	const char *op;
+
+	if (mgmt_event == NET_EVENT_DNS_SERVER_ADD) {
+		op = "Added";
+	} else {
+		op = "Deleted";
+	}
+	if (entry->sa_family == AF_INET) {
+		net_addr_ntop(AF_INET, &net_sin(entry)->sin_addr, addr, sizeof(addr));
+	} else if (entry->sa_family == AF_INET6) {
+		net_addr_ntop(AF_INET6, &net_sin6(entry)->sin6_addr, addr, sizeof(addr));
+	} else {
+		strcpy(addr, "<unknown format>");
+	}
+	printf("%s DNS Address: %s\n", op, addr);
+}
+
+static void ota_handle_wifi_scan_result(struct net_mgmt_event_callback *cb)
+{
+	const struct wifi_scan_result *entry = (const struct wifi_scan_result *)cb->info;
+	uint8_t mac_string_buf[sizeof("xx:xx:xx:xx:xx:xx")];
+	uint8_t ssid_print[WIFI_SSID_MAX_LEN + 1];
+	struct app_ctx *ctx = CONTAINER_OF(cb, struct app_ctx, wlan_mgmt_cb);
+
+	if (strncmp(entry->ssid, CONFIG_OTA_WIFI_SSID, entry->ssid_length) == 0) {
+		ctx->security_type = entry->security;
+	}
+
+	if (!ctx->scan_result_count) {
+		printf("%-4s | %-32s %-5s | %-13s | %-4s | %-15s | %-17s | %-8s\n", "Num", "SSID",
+		       "(len)", "Chan (Band)", "RSSI", "Security", "BSSID", "MFP");
+	}
+
+	strncpy(ssid_print, entry->ssid, sizeof(ssid_print) - 1);
+	ssid_print[sizeof(ssid_print) - 1] = '\0';
+
+	printf("%-4d | %-32s %-5u | %-4u (%-6s) | %-4d | %-15s | %-17s | %-8s\n",
+	       ++ctx->scan_result_count, ssid_print, entry->ssid_length, entry->channel,
```

Comment:
Honestly, numbering the results do not bring any benefit. In the other hand it complexify the code. I suggest to just drop it.